### PR TITLE
API 7.6 - `PaidMedia*` classes

### PR DESCRIPTION
**Pull request by harshil21**
### Checklist

- [x] Added ``.. versionadded:: NEXT.VERSION``, ``.. versionchanged:: NEXT.VERSION`` or ``.. deprecated:: NEXT.VERSION`` to the docstrings for user facing changes (for methods/class descriptions, arguments and attributes)
- [x]  Created new or adapted existing unit tests
- [ ] Documented code changes according to the `CSI standard <https://standards.mousepawmedia.com/en/stable/csi.html>`__
- [ ] Added myself alphabetically to ``AUTHORS.rst`` (optional)
- [x] Added new classes & modules to the docs and all suitable ``__all__`` s
- [ ] Checked the `Stability Policy <https://docs.python-telegram-bot.org/stability_policy.html>`_ in case of deprecations or changes to documented behavior

**If the PR contains API changes (otherwise, you can ignore this passage)**

- [x] Checked the Bot API specific sections of the [Stability Policy](https://docs.python-telegram-bot.org/stability_policy.html)

- New classes:

   - [x] Added ``self._id_attrs`` and corresponding documentation
   - [x] ``__init__`` accepts ``api_kwargs`` as kw-only

- Added new shortcuts:

   - [ ]  In :class:`~telegram.Chat` & :class:`~telegram.User` for all methods that accept ``chat/user_id``
   - [ ]  In :class:`~telegram.Message` for all methods that accept ``chat_id`` and ``message_id``
   - [ ]  For new :class:`~telegram.Message` shortcuts: Added ``do_quote`` argument if methods accepts ``reply_to_message_id``
   - [ ] In :class:`~telegram.CallbackQuery` for all methods that accept either ``chat_id`` and ``message_id`` or ``inline_message_id``

-  If relevant:

   - [x] Added new constants at :mod:`telegram.constants` and shortcuts to them as class variables
   - [x] Link new and existing constants in docstrings instead of hard-coded numbers and strings
   - [ ]  Add new message types to :attr:`telegram.Message.effective_attachment`
   - [ ] Added new handlers for new update types
   - [ ] Add the handlers to the warning loop in the :class:`~telegram.ext.ConversationHandler`
   - [ ] Added new filters for new message (sub)types
   - [x]  Added or updated documentation for the changed class(es) and/or method(s)
   - [ ] Added the new method(s) to ``_extbot.py``
   - [ ] Added or updated ``bot_methods.rst``
   - [ ] Updated the Bot API version number in all places: ``README.rst`` and ``README_RAW.rst`` (including the badge), as well as ``telegram.constants.BOT_API_VERSION_INFO``
   - [ ] Added logic for arbitrary callback data in :class:`telegram.ext.ExtBot` for new methods that either accept a ``reply_markup`` in some form or have a return type that is/contains :class:`~telegram.Message`

### CODE REVIEW

File: telegram/_paidmedia.py [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+
+class PaidMedia(TelegramObject):
+    """Describes the paid media added to a message. Currently, it can be one of:
+
+    * :class:`telegram.PaidMediaPreview`
+    * :class:`telegram.PaidMediaPhoto`
+    * :class:`telegram.PaidMediaVideo`
+
+    Objects of this class are comparable in terms of equality. Two objects of this class are
+    considered equal, if their :attr:`type` are equal.
```
Comment (Bibo-Joshi) [inline telegram/_paidmedia.py:43]:
```suggestion
    considered equal, if their :attr:`type` is equal.
```

File: telegram/_paidmedia.py [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+
+        Args:
+            data (Dict[:obj:`str`, ...]): The JSON data.
+            bot (:class:`telegram.Bot`, optional): The bot associated with this object.
+
+        Returns:
+            The Telegram object.
+
+        """
+        data = cls._parse_data(data)
```
Comment (Bibo-Joshi) [inline telegram/_paidmedia.py:90]: just an idea: we use this same logic now in several classes. We could think of extracting this logic to a helper method, possibly a protected method in TelegramObject.

File: tests/test_stars.py
Comment (Bibo-Joshi) [inline tests/test_stars.py:1]: are these actual changes in API 7.6 or should that rather be a test-fix follow-up for API 7.5?

**Review by Bibo-Joshi — COMMENTED** [review]
🚀

File: tests/test_stars.py
Comment (harshil21) [reply tests/test_stars.py:1]: they are a test fix for api 7.5

File: telegram/_paidmedia.py (hunk 2 above)
Comment (harshil21) [reply telegram/_paidmedia.py:90]: we still have to specify the `_class_mapping` on every call isn't it? which parts / how exactly do you want to extract this?

Comment (Bibo-Joshi) [reply telegram/_paidmedia.py:90]: I was thinking something like

```python
def de_json(…):
    super()._de_json_subclasses(PaidMedia, {cls.PREVIEW: …})

def _de_json_subclasses(cls, data, base_class, class_mapping):
    data = cls._parse_data(data)
     if data is None:
            return None

        if not data and cls is base_class:
            return None

        class_mapping: Dict[str, Type[PaidMedia]] = {
            cls.PREVIEW: PaidMediaPreview,
            cls.PHOTO: PaidMediaPhoto,
            cls.VIDEO: PaidMediaVideo,
        }

        if cls is PaidMedia and data.get("type") in class_mapping:
            return class_mapping[data.pop("type")].de_json(data=data, bot=bot)

        return super().de_json(data=data, bot=bot)
```

But just a thought - doesn't have to be in this PR or at all

File: tests/test_stars.py
Comment (Bibo-Joshi) [reply tests/test_stars.py:1]: Then I prefer having a standalone PR for that

File: telegram/_paidmedia.py (hunk 2 above)
Comment (harshil21) [reply telegram/_paidmedia.py:90]: yeah, probably better to do in a different PR so we can apply to other classes

Comment (Bibo-Joshi) [reply telegram/_paidmedia.py:90]: if you have the time, I would be glad to see an issue for it 😅

**Pull request merged**